plugins/labormanager/labormanager.cpp

```diff
@@ -1518,6 +1519,26 @@ class AutoLaborManager {
             else if (altruism <= 24)
                 score -= 50000;
         }
+
+        // Favor/disfavor BUTCHER (covers slaughtering), HAUL_ANIMALS (covers caging), and CUTWOOD based on NATURE value
+
+        if (labor == df::unit_labor::BUTCHER || labor == df::unit_labor::HAUL_ANIMALS || labor == df::unit_labor::CUTWOOD)
+        {
+            int nature = 0;
+            for (auto i = d->dwarf->status.current_soul->personality.values.begin();
+                i != d->dwarf->status.current_soul->personality.values.end();
+                i++)
+            {
```

Comment:
Another possible range-based-for use case, a la #1380
(I know this plugin predates our move to C++11-supporting compilers, so it's not really an issue.)